internal/service/auth.go

```diff
@@ -1,61 +1,262 @@
 package service
 
 import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/golang-jwt/jwt/v5"
+	"github.com/jackc/pgx/v5"
+	"github.com/vultisig/verifier/internal/storage"
+	"github.com/vultisig/verifier/internal/types"
 )
 
 type Claims struct {
 	jwt.RegisteredClaims
+	PublicKey string `json:"public_key"`
+	TokenID   string `json:"token_id"`
 }
 
 const (
 	expireDuration = 7 * 24 * time.Hour
+	tokenIDLength  = 32
 )
 
 type AuthService struct {
 	JWTSecret []byte
+	db        storage.DatabaseStorage
 }
 
 // NewAuthService creates a new authentication service
-func NewAuthService(secret string) *AuthService {
+func NewAuthService(secret string, db storage.DatabaseStorage) *AuthService {
 	return &AuthService{
 		JWTSecret: []byte(secret),
+		db:        db,
 	}
 }
 
-// GenerateToken creates a new JWT token
-func (a *AuthService) GenerateToken() (string, error) {
+// generateTokenID generates a random token ID
+func generateTokenID() (string, error) {
+	b := make([]byte, tokenIDLength)
+	if _, err := rand.Read(b); err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(b), nil
+}
+
+// GenerateToken creates a new JWT token and stores it in the database
+func (a *AuthService) GenerateToken(publicKey string) (string, error) {
+	ctx := context.Background()
+
+	// Generate a unique token ID
+	tokenID, err := generateTokenID()
+	if err != nil {
+		return "", err
+	}
+
 	expirationTime := time.Now().Add(expireDuration)
 	claims := &Claims{
 		RegisteredClaims: jwt.RegisteredClaims{
 			ExpiresAt: jwt.NewNumericDate(expirationTime),
 			IssuedAt:  jwt.NewNumericDate(time.Now()),
 		},
+		PublicKey: publicKey,
+		TokenID:   tokenID,
 	}
+
+	// Create JWT token
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
-	return token.SignedString(a.JWTSecret)
+	tokenString, err := token.SignedString(a.JWTSecret)
+	if err != nil {
+		return "", err
+	}
+
+	// Store token in database
+	tx, err := a.db.Pool().Begin(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer tx.Rollback(ctx)
```

Comment:
refer to `a.db.CreateVaultToken` , it doesn't need a transaction , so you can remove all these